zipline-gradle-plugin/src/main/kotlin/app/cash/zipline/gradle/ZiplinePlugin.kt

```diff
@@ -95,4 +99,21 @@ class ZiplinePlugin : KotlinCompilerPluginSupportPlugin {
       listOf() // No options.
     }
   }
+
+  private fun createGenerateKeyPairTask(project: Project) {
+    val name = "generateZiplineManifestKeyPair"
+    val rootProjectTasks = project.rootProject.tasks
+    if (rootProjectTasks.findByName(name) == null) {
+      rootProjectTasks.create(name) { task ->
```

Comment:
Plugins can only be applied once. There's no need to do this conditional create.

Also `create` should be `register` in order to defer the task configuration when not invoked.

Reply:
I’m cheating here and adding a task to the root project even if the plugin is installed on a subproject. Does that change this advice?

Reply:
Ah. Yes and no!

So yes because then doing it lazy is required. But no, because project isolation (a forthcoming Gradle feature) will actively prevent you from doing this.

If you run `gw taskName` and the task only exists in `:foo:bar:baz:taskName` it will still run correctly. So it should be safe to add the task only to the project to which the plugin is applied.

Reply:
Ahhh, I’ll define it in the right place.